From 3d52e8f7382eae8896b36a9b24c3ed42622afd9d Mon Sep 17 00:00:00 2001
From: "Song, Yoong Siang" <yoong.siang.song@intel.com>
Date: Tue, 1 Sep 2020 20:30:33 +0800
Subject: [PATCH 3/5] Revert "REVERTME: net: stmmac: use DMA Bus Mode INTM MODE
 0"

Reverting this commit as PSE GbE do not have intr missing issue.

This reverts commit f1c34f67cff5c2eb1cd1622e04f7c995f2c31590.

Signed-off-by: Song, Yoong Siang <yoong.siang.song@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c b/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c
index cfdd936..365f3e9 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac4_dma.c
@@ -165,8 +165,10 @@ static void dwmac4_dma_init(void __iomem *ioaddr,
 
 	value = readl(ioaddr + DMA_BUS_MODE);
 
-	value &= ~DMA_BUS_MODE_INTM_MASK;
-	value |= (DMA_BUS_MODE_INTM_MODE0 << DMA_BUS_MODE_INTM_SHIFT);
+	if (dma_cfg->multi_msi_en) {
+		value &= ~DMA_BUS_MODE_INTM_MASK;
+		value |= (DMA_BUS_MODE_INTM_MODE1 << DMA_BUS_MODE_INTM_SHIFT);
+	}
 
 	writel(value, ioaddr + DMA_BUS_MODE);
 }
-- 
2.7.4

